foxy-sros2 migration notes (#1)
mikaelarguedas/ros2_documentation: source/Releases/Release-Foxy-Fitzroy.rst

- ``std_msgs/msg/UInt8``
- ``std_msgs/msg/UInt8MultiArray``

> sros2

Reviewer: This was a fairly wide-ranging change that wasn't just sros2. Can we just say:

```suggestion
Security features
```

instead?

Author: I was wondering the same thing. Most sections seem to be package-specific but there are some non package-specific sections as well, so let's rename the section and see what the reviewers say

> ^^^^^

Author: ```suggestion
^^^^^^^^^^^^^^^^^
```

Use of security enclaves
""""""""""""""""""""""""

> As of Foxy, domain participants are not mapped directly to ROS nodes anymore.
> The concept of security enclave was introduced, an enclave is a process or group of processes that will share the same identity and access control rules.

Reviewer: Slight rewording, addition of some details, and removal of some passive voice:

```suggestion
As of Foxy, domain participants are no longer mapped directly to ROS nodes.
As a result, ROS 2 security features (which are specific to domain participants) are also no longer mapped directly to ROS nodes.
Instead, Foxy introduces the concept of a security "enclave", where an "enclave" is a process or group of processes that will share the same identity and access control rules.
```

> Each process using ROS Security is part of an enclave.

Reviewer: It's not entirely clear what you mean by this statement, can you clarify?

Author: I believe this one can be removed if we keep the next one.
My goal was to say that processes using security will always load an enclave's security material and that the enclave name need to be specified (as opposed to infer from node name and namespace

Reviewer: I'm fine on the next one, should we remove this one then?

> This means that security artifacts are **not** retrieved based on the node name anymore but based on the Security enclave name.

Reviewer: Assuming you like my edit above, we can probably get rid of this sentence.

```suggestion
```

Author: My goal here was to highlight what it actually changes for users. The paragraph above is generic but it's unclear from the outside what concrete change a user need to perform to migrate a secured system from Eloquent to Foxy. E.g. it doesnt seem too far-fatched for a user to assume that by default the nodes would use an enclave matching the node name

Reviewer: Okay fair enough.

A node enclave name can be set by using the ROS argument `--enclave`, e.g. `ros2 run demo_nodes_py talker --ros-args --enclave /my_enclave`

Related design document: https://github.com/ros2/design/pull/274

Renaming of the environment variables
"""""""""""""""""""""""""""""""""""""

.. list-table:: Environment variables renaming
   :widths: 25 25
   :header-rows: 1

   * - Name in Eloquent

>      - Name in Foxy

Reviewer: Good idea on this, I like it (although note that I don't know table formats in rst so I'll just assume it's correct!).

Author: Discovering as I go :smile: gave it a try at http://rst.ninjs.org/ and it seems to work so I'll assume it'll look ok on the index website as well
